### PR TITLE
Configuration Fixes

### PR DESCRIPTION
**Pull request by zzzeid**
**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: strawberryfields/configuration.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
 
    * On Linux: ``~/.config/strawberryfields``
    * On Windows: ``~C:\Users\USERNAME\AppData\Local\Xanadu\strawberryfields``
-   * On MacOS: ``~/Library/Preferences/strawberryfields``
+   * On MacOS: ``~/Library/Application\ Support/strawberryfields``
```
Comment (josh146) [inline strawberryfields/configuration.py]: Thanks for fixing this @zzzeid :+1:

File: strawberryfields/configuration.py [hunk 2]
```diff
@@ -160,7 +160,7 @@ def update_config(self):
                 if env in os.environ:
                     # Update from environment variable
                     self._config[section][key] = os.environ[env]
-                elif self._config_file:
+                elif self._config_file and key in self._config_file[section]:
```
Comment (co9olguy) [inline strawberryfields/configuration.py]: this feels like something that should lead to a new test maybe?

File: strawberryfields/api_client.py [hunk 3]
```diff
@@ -141,37 +141,36 @@ class APIClient:
...
+        "localhost:8080",
+        "platform.strawberryfields.ai",
+    ]
 
     DEFAULT_HOSTNAME = "localhost"
 
-    ENV_KEY_PREFIX = "SF_API_"
-    ENV_AUTHENTICATION_TOKEN_KEY = "{}AUTHENTICATION_TOKEN".format(ENV_KEY_PREFIX)
-    ENV_API_HOSTNAME_KEY = "{}API_HOSTNAME".format(ENV_KEY_PREFIX)
-    ENV_USE_SSL_KEY = "{}USE_SSL".format(ENV_KEY_PREFIX)
```
Comment (co9olguy) [inline strawberryfields/api_client.py]: Were these not used anywhere else?

File: strawberryfields/configuration.py (hunk 2 above)
Comment (zzzeid) [reply strawberryfields/configuration.py]: Yes - this line is to fix the `KeyError` issue that we got when we excluded a parameter from the config file (i.e. the file was behaving as though all parameters were required). I can add a test for this.

File: strawberryfields/api_client.py (hunk 3 above)
Comment (zzzeid) [reply strawberryfields/api_client.py]: No they are not. These were left behind after a previous refactor.

File: strawberryfields/configuration.py (hunk 2 above)
Comment (zzzeid) [reply strawberryfields/configuration.py]: Test added.

**Pull request merged**